simplekv/db/sql.py

```diff
@@ -44,6 +45,18 @@ def _get(self, key):
     def _open(self, key):
         return BytesIO(self._get(key))
 
+    def _move(self, source, dest):
+        if not self._has_key(source):
+            raise KeyError(source)
+        con = self.bind.connect()
+        with con.begin():
+            # delete the potential existing previous key
+            con.execute(self.table.delete(self.table.c.key == dest))
+            con.execute(self.table.update().values(key=dest).where(self.table.c.key == source))
```

Comment:
This is what (supposedly?) happens:
if we do move(source, dest) we want overwrite semantics if dest already exists.
As each entry is a distinct row, we first drop a potentially existing entry with key=dest, then we execute the move by changing the key of the entry with key=source to dest.
All of this happens in a transaction (if supported by the database), so this is an atomic update.